sys/phydat/phydat_str.c

```diff
@@ -32,10 +32,27 @@ void phydat_dump(phydat_t *data, uint8_t dim)
     }
     printf("Data:");
     for (uint8_t i = 0; i < dim; i++) {
-        char tmp[PHYDAT_SCALE_STR_MAXLEN];
-        phydat_scale_to_str(data->scale, tmp);
-        printf("\t[%i] %i%s%s\n", (int)i, (int)data->val[i], tmp,
-               phydat_unit_to_str(data->unit));
+        char scale_str = phydat_scale_to_str(data->scale);
+
+        printf("\t[%i] ", (int)i);
+
+        if (scale_str) {
+            printf("%i%c", (int)data->val[i], scale_str);
+        }
+        else if (data->scale == 0) {
+            printf("%i", (int)data->val[i]);
+        }
+        else if ((data->scale > -6) && (data->scale < 0)) {
+            char num[8];
+            size_t len = fmt_s16_ffp(num, data->val[i], data->scale * -1);
```

Comment:
could be an option, but this is unrelated to this PR. In this particular case above I am using fmt, as stdio does not provide the feature I need... (or does it?)